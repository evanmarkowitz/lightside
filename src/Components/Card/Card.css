.card {
  border: 1px solid black;
  margin: 10px;
  height: 300px;
  width: 300px;
  padding: 10px;
  background: rgba(63, 100, 131, 0.38);
}

.card-header {
  display: flex;
  justify-content: space-between
}

.card-image {
  height: 50px;
  width: 50px;
}