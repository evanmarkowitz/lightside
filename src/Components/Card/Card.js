import React from 'react'
import './Card.css'
import rebel from '../../images/rebel-2.png'
import empire from '../../images/Alliance_of_free_planets.png'
import PropTypes from 'prop-types'

const Card = ({data, id, toggleFavorite, category, isFavorited}) => {
  const showCards = Object.entries(data).map(entry => {
    let key = entry[0].replace('_', ' ')
    return <li 
      className='card-list-item' 
      key={entry[0]}>
      <span className="card-key">{key}:</span> <span className="card-entry">{entry[1]}</span>
      </li>
  })
  
  return(
    <article className="card" id={id}>
      <div className="card-header">
      <h2 className="card-title">{data.name}</h2>
      <img 
        src={isFavorited ? rebel : empire} 
        className='card-image' 
        onClick={() => toggleFavorite(data.name, category)}
        alt="dumb dumb"/>
      </div>
      <ul className="card-list">
      {showCards}
      </ul>
    </article>
  )

}

Card.propTypes = {
  data: PropTypes.object,
  id: PropTypes.string,
  toggleFavorite: PropTypes.func,
  category: PropTypes.string,
  isFavorited: PropTypes.bool
}

export default Card