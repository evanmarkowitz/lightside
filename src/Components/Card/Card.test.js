import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from './Card.js';

describe('Card', () => {
  let fakeData;
  let toggleFavorite;
  beforeEach(() => {
      fakeData = {
      diameter: "10200",
      name: "Yavin IV",
      population: "1000", 
      terrain: "jungle, rainforests"
    }
    toggleFavorite= jest.fn()
  });
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow( <Card
      data = {fakeData}
      id= "card1"
      isFavorited = {false}
      toggleFavorite={jest.fn()}
    />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot with isFavorited true passed in correctly', () => {
    const wrapper = shallow( <Card
      data = {fakeData}
      id= "card1"
      isFavorited = {true}
      toggleFavorite={jest.fn()}
    />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should call the toggleFavorite function when the card-image is clicked', () => {
    const wrapper = shallow( <Card
      data = {fakeData}
      id= "card1"
      isFavorited = {true}
      toggleFavorite={toggleFavorite}
      category = 'vehicle'
    />);
    wrapper.find('.card-image').simulate('click');
    expect(toggleFavorite).toBeCalledWith("Yavin IV", 'vehicle');
  });

  
});