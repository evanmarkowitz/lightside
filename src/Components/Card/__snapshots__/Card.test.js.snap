// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `ShallowWrapper {}`;

exports[`Card should match the snapshot with isFavorited true passed in correctly 1`] = `ShallowWrapper {}`;
